content/en/blog/falcosidekick-reponse-engine-part-5-argo.md

```diff
@@ -0,0 +1,377 @@
+                  parameters:
+                    - name: falco-event
+                      value: {}
+EOF
+```
+
+## Creation of the Workflow Template
+
+One piece is missing in our `Argo` stack, we mentionned a template above, we logically need to create it too, with the service account it needs.
+
+```shell+
+cat <<EOF | kubectl apply -n argo -f -
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: delete-pod-sa
+  namespace: argo
+---
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: delete-pod-sa-cluster-role
+rules:
+  - apiGroups: [""]
+    resources: ["pods"]
+    verbs: ["get", "list", "delete", "patch", "watch"]
+---
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: delete-pod-sa-cluster-role-binding
+roleRef:
+  kind: ClusterRole
+  name: delete-pod-sa-cluster-role
+  apiGroup: rbac.authorization.k8s.io
+subjects:
+  - kind: ServiceAccount
+    name: delete-pod-sa
+    namespace: argo
+EOF
+```
+```shell+
+cat <<EOF | kubectl apply -n argo -f -
+apiVersion: argoproj.io/v1alpha1
+kind: WorkflowTemplate
+metadata:
+  name: delete-pod-template
+  namespace: argo
+spec:
+  entrypoint: delete-pod
+  serviceAccountName: delete-pod-sa
+  arguments:
+    parameters:
+      - name: falco-event
+        value: "{}"
+  templates:
+    - name: delete-pod
+      inputs:
+        parameters:
+          - name: falco-event
+      container:
+        image: devopps/kubernetes-response-engine-based-on-event-driven-workflow@sha256:22ee203a33fe88f0f99968daebdcea0ca52c8a3d6f7af4c823ed78ac15b7c5db
+        env:
+          - name: BODY
+            value: "{{inputs.parameters.falco-event}}"
+EOF
+```
+
+`Argo Workflow` runs all workflow steps inside their own pods, we'll use for this tutorial a *Golang* image developped by [@developer-guy](https://github.com/developer-guy) (who wrote the [Part 2]({{< ref "/blog/falcosidekick-reponse-engine-part-2-openfass" >}}) of this serie 😄).
+
+
+At this stage, everything is ready to receive events from `Falco` and protect our cluster.
+If you go in `Argo Workflow` UI you will find the architecture we described at beginning.
+
+![Event Flow for our Response Engine](/img/argo-workflow-response-engine-event-flow.png)
+
+## Installation of Falco and Falcosidekick
+
+Last but not least, it's time to install our beloved `Falco` and `Falcosidekick` and connect them to our shiny new Response Engine.
+
+Like for other posts of this serie we'll use `Helm` as conveniant installation method.
+
+```shell+
+kubectl create ns falco
+
+helm repo add falcosecurity https://falcosecurity.github.io/charts
+
+helm install falco falcosecurity/falco \
+  --namespace falco \
+  --set falcosidekick.enabled=true \
+  --set falcosidekick.config.webhook.address=http://webhook-falco-eventsource-svc.argo-events.svc.cluster.local:12000/falco
+```
+
+You remember the service we mentionned earlier? This is it we use, in its FQDN format, as endpoint.
```

Comment:
Remember the service we "mentioned" earlier? This is it in its FQDN format as an endpoint.